name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/PDFParserTest.java

```diff
@@ -0,0 +1,136 @@
+package name.abuchen.portfolio.datatransfer.pdf;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.regex.Pattern;
+
+import org.junit.Test;
+
+import name.abuchen.portfolio.datatransfer.DocumentContext;
+
+@SuppressWarnings("nls")
+public class PDFParserTest
+{
+    @Test
+    public void testMatches_Type1()
```

Comment:
I do not understand what is meant by "type1" through "type5" - can you shed some light?

Also, the test "type1" and "type2" tests look pretty identical to me - the comment is different, but I do not see a difference in the code. They all put a strings in a map and retrieve them again. Am I overlooking something?

To be honest, I would not bother writing specific tests for the PDFDocument in isolation. **All** PDF tests are testing this functionally implicitly. I cannot image a situation where all PDF tests are green, but this test is failing. And the PDF tests are better because they test on the level of business functionality.

Reply:
done... i delete this file